File: java/src/main/java/sa/amnn/kit/resources/Withdrawals.java
package sa.amnn.kit.resources;

import com.fasterxml.jackson.core.type.TypeReference;
import java.util.Map;
import sa.amnn.kit.AmenClient;
import sa.amnn.kit.Models.Page;
import sa.amnn.kit.Models.Withdrawal;

public final class Withdrawals {
    private final AmenClient c; public Withdrawals(AmenClient c) { this.c = c; }
    public Withdrawal create(String bankAccountId, String amount) { return c.post("/withdrawals/", Map.of("bank_account_id", bankAccountId, "amount", amount), new TypeReference<>() {}); }
    public Withdrawal get(String n) { return c.get("/withdrawals/" + n, Map.of(), new TypeReference<>() {}); }
    public Page<Withdrawal> list(Map<String, String> params) { return c.page("/withdrawals/", params, "withdrawals", Withdrawal.class); }
}

File: csharp/src/Amen.Kit/Resources/Withdrawals.cs
namespace Amen.Kit.Resources;
public sealed class Withdrawals(AmenClient c)
{
    public async Task<Withdrawal> CreateAsync(string bankAccountId, string amount) => (await c.PostAsync<Withdrawal>("/withdrawals/", new { bank_account_id = bankAccountId, amount }))!;
    public async Task<Withdrawal> GetAsync(string n) => (await c.GetAsync<Withdrawal>($"/withdrawals/{n}"))!;
    public Task<Page<Withdrawal>> ListAsync(IDictionary<string, string?>? query = null) => c.PageAsync<Withdrawal>("/withdrawals/", query, "withdrawals");
}
